pypsa/io.py

```diff
@@ -1010,7 +1009,14 @@ def import_from_pandapower_net(network, net, extra_line_data=False):
     >>> network.import_from_pandapower_net(net, extra_line_data=True)
     """
     logger.warning("Warning: Importing from pandapower is still in beta; not all pandapower data is supported.\nUnsupported features include: three-winding transformers, switches, in_service status, shunt impedances and tap positions of transformers.")
-
+    
+    if use_pandapower_index:
+        for element in ["bus", "line", "gen", "load", "shunt", "trafo", "switch"]:
+            net[element].name = net[element].index.values
+        
+        net.sgen.name = net.sgen.index.values + (net.gen.index.max() + 1 if len(net.gen) else 0)
+        net.ext_grid.name = net.ext_grid.index.values + (net.gen.index.max() + 1 if len(net.gen) else 0) + (net.sgen.index.max() + 1 if len(net.sgen) else 0)
```

Comment:
not sure I understand what this segment does.

Reply:
This segment takes care of unique naming of all assets. Pandapower splits sgen, gen and ext_grid (slack generator). To guarantee unique bus names required naming changes in pandapower. Line 1017-1018 just add a running number to the name.

Reply:
This is a bit hacky. You change the dataframes of the pandapower network which should not happen (the pandapower network is only a input and should leave the function as it came in). Do you really think this is necessary?

Reply:
removed these lines